src/model/model.rs

```diff
@@ -46,8 +49,40 @@ impl std::convert::From<Vec<u8>> for Block {
     }
 }
 
-#[derive(Clone, Debug, serde::Serialize, serde::Deserialize)]
-pub struct Hash {}
+#[derive(Clone, Debug, PartialEq)]
+pub struct Hash(GenericArray<u8, U64>);
```

Comment:
Great struct (I'm talking about `GenericArray`).
Maybe we can make our `Hash` a type alias for `GenericArray` instead of struct?

Reply:
I think implementing traits for type aliases is still in nightly. How would you advise to implement serialization then?

Reply:
oh, I see now `Vec::<u8>::serialize(&self.0.to_vec()` - okay, maybe you may create an issue to make a PR into lib with serializer?

Reply:
Yes, that would make sense